stan/math/prim/err/check_cholesky_factor.hpp

```diff
@@ -25,17 +28,41 @@ namespace math {
  *   factor, if number of rows is less than the number of columns,
  *   if there are 0 columns, or if any element in matrix is NaN
  */
-template <typename EigMat, require_eigen_t<EigMat>* = nullptr>
+template <typename Mat, require_matrix_t<Mat>* = nullptr>
 inline void check_cholesky_factor(const char* function, const char* name,
-                                  const EigMat& y) {
+                                  const Mat& y) {
   check_less_or_equal(function, "columns and rows of Cholesky factor", y.cols(),
                       y.rows());
   check_positive(function, "columns of Cholesky factor", y.cols());
-  const Eigen::Ref<const plain_type_t<EigMat>>& y_ref = y;
+  auto&& y_ref = to_ref(value_of_rec(y));
   check_lower_triangular(function, name, y_ref);
   check_positive(function, name, y_ref.diagonal());
 }
 
+/**
+ * Check if the specified matrix is a valid Cholesky factor.
+ * A Cholesky factor is a lower triangular matrix whose diagonal
+ * elements are all positive.  Note that Cholesky factors need not
+ * be square, but require at least as many rows M as columns N
+ * (i.e., M &gt;= N).
+ * @tparam StdVec A standard vector with inner type inheriting from `MatrixBase`
```

Comment:
We only have a `check_cholesky_factor` that works for this definition so if someone tried passing something like a `std::vector<std::vector<double>>` it would just fail to compile